woocommerce/woocommerce-ios
# Disallow entering discounts higher than price

Closes: #14080


## Description
There's a discrepancy between platforms regarding how we allow discounts to be applied to products, in iOS we allow discounts higher than the product price, while Android does not. 

With this PR we get closer to Android by not allowing these, displaying an error message when a merchant attempts to do so, and disallowing the `Add` button of working until is resolved.

![Simulator Screen Recording - iPhone 15 - 2024-10-01 at 11 22 59](https://github.com/user-attachments/assets/1416be44-68f7-4aca-a65c-adaaf1fb55f6)

There are further UI and logic improvements that can be done in this screen, but I have left them outside this PR to keep a reduced scope (eg: The "remove discount" button appears when there are no discounts saved yet).

Note: There seems to be a lot of unnecessary complexity in these discount-related screens, most likely because we haven't updated the logic from when we stopped handling cases that no longer exist (fees). At some point would be good to separate "discounts" from "legacy fees", and improve the general design of how we apply these.

## Testing
* Go to `Orders` > `+` > `+ Add product` > expand the product and tap on `+ Add discount`
* Observe that if you enter a fixed-type discount higher than the product price:
  * An error message appears
  * The `Add` button is disabled
  * Dismissing the modal does not save changes
 * Repeat the same for a percentage-type discount and observe the same behaviour if the discount is higher than 100%.

| Fixed | Percentage |
|--------|--------|
| ![simulator_screenshot_976DD458-DE2C-4C15-9854-74210830F7AC](https://github.com/user-attachments/assets/634076d9-b793-4574-b981-5122fd8d4051) | ![simulator_screenshot_FB0F5C9E-9987-43EB-A02B-5BCADF5DA1E5](https://github.com/user-attachments/assets/7cbdf80d-23b4-4799-9266-c778d2ffd7d0) | 
---


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
